### PR TITLE
fix(293): meta get enable to get map and array meta

### PR DESCRIPTION
**Pull request by tk3fftk**
### Before
`meta get` could not get nested value like following:
```
sh-3.2$ ./meta set foo.bar baz
sh-3.2$ ./meta get foo && echo
{"bar":"baz"}
sh-3.2$ ./meta get foo.bar && echo
null
```

### After
I fixed to enable to get such value like following:
```
$ ./meta set foo.bar baz                                                                                                                                           
$ ./meta get foo                                                                                                                                                   
{"bar":"baz"}%         
$ ./meta get foo.bar                                                                                                                                                                                                                                                                                                                         
baz%
```

and

```
$ ./meta set foo[2].bar baz
$ ./meta get foo                                                                                                                                                      
[null,null,{"bar":"baz"}]%
$ ./meta get foo[2]
{"bar":"baz"}%
$./meta get foo[2].bar
baz
```


I have no good idea about function naming😥 
If you have idea, please teach me on comment.

Related: https://github.com/screwdriver-cd/screwdriver/issues/293#issuecomment-286246365

### CODE REVIEW

**Comment by tk3fftk** [comment]
@minz1027 Thank you for review! I'll do the refactor next PR😃

File: meta.go [hunk 1]
```diff
@@ -54,6 +56,82 @@ func getMeta(key string, metaSpace string, output io.Writer) error {
 	return nil
 }
 
+func fetchMetaValue(key string, meta interface{}) (string, interface{}) {
+	var result interface{}
+	for position, char := range key {
```
Comment (d2lam) [inline meta.go]: rename `position` to `current` to make it more clear

File: meta.go [hunk 2]
```diff
@@ -54,6 +56,82 @@ func getMeta(key string, metaSpace string, output io.Writer) error {
...
+				}
+			}
+			metaIndex, _ := strconv.Atoi(key[position+1 : i]) // e.g. if array[10], get "10"
+			shortenKey := key[i+1:]                           // Remove bracket[num] from key
+			metaValue := reflect.ValueOf(meta)
+			// convert type interface -> Value -> map[string]interface{}
+			metaMap := make(map[string]interface{})
+			if metaValue.Kind() == reflect.Map {
+				for _, k := range metaValue.MapKeys() {
+					pk, _ := k.Interface().(string)
```
Comment (d2lam) [inline meta.go]: Rename `pk`? What does it stand for?

Comment (tkyi) [inline meta.go]: PK FIRE! 🔥 
PK fire

Comment (tk3fftk) [inline meta.go]: Maybe it means previous Meta key but `k` is also previous Meta key... It is only type conversion.

File: meta.go (hunk 1 above)
Comment (tk3fftk) [inline meta.go]: fixed.

File: meta.go (hunk 2 above)
Comment (tk3fftk) [inline meta.go]: I fixed these:
k -> kv (type of key is Value)
pk -> ps (type of key is string)

Comment (d2lam) [inline meta.go]: Maybe rename to `keyValue` and `keyString`?

Comment (tk3fftk) [inline meta.go]: Fixed!!

**Pull request merged**